doc.go

```diff
@@ -32,6 +32,8 @@
 // In this case, the string "sha256" is the algorithm and the hex bytes are
 // the "digest".
 //
+// This package exclusively uses hex encoding for all digests.
```

Comment:
IMO this wording is a bit confusing, but I've posted a revised take @ d1caf20, PTAL